### DISCUSSION TITLE
maturin failed, Caused by: failed to list package files

### DISCUSSION THREAD

**Discussion by ClementWalter** [PyO3/maturin#2388] [Q&A] [answered]
I'm just starting with maturin and doing the init code

```
maturin init -b pyo3 --mixed
maturin develop --uv
```

fails with 

```
🍹 Building a mixed python/rust project
🔗 Found pyo3 bindings
🐍 Found CPython 3.10 at /Users/clementwalter/Documents/string_sum/.venv/bin/python
📡 Using build options features from pyproject.toml
Audited 2 packages in 0.48ms
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
📦 Built wheel for CPython 3.10 to /var/folders/7y/kppn0q7946gczmpmnlc5bncm0000gn/T/.tmphpDGPx/string_sum-0.1.0-cp310-cp310-macosx_11_0_arm64.whl
✏️  Setting installed package as editable
💥 maturin failed
  Caused by: failed to list package files
```

I've tried to add more logs with `-vvv` but it doesn't display any information on the error.

The code is https://github.com/ClementWalter/string_sum

**Comment by ClementWalter** [comment]
I have tried to investigate a bit more and I'm wondering if it's not related to how uv manages pip. See also https://github.com/astral-sh/uv/issues/6593.

Adding `uv add pip`, `maturin develop --uv` still fails but `maturin develop` works (!).

Note that in any cases (even when it fails), `maturin build` does work, and the packaging is ready, since the test passes

**Reply by messense** [reply]
What's your `uv` version? I think `maturin develop --uv` currently requires uv `0.4.25` and later.

The error comes from here https://github.com/PyO3/maturin/blob/5342a834a53b7ee773b76202d65157da4f24d10f/src/develop.rs#L273

cc @mbway

**Reply by mbway** [reply, marked as answer]
Yeah, old UV doesn't support `pip show --files`

Should I put a check in place that gives a more informative error if a UV is used that is too old?

**Reply by messense** [reply]
That'd be great, but IMO a simpler solution is just change the error message to say that we require uv >= 0.4.25.

**Reply by ClementWalter** [reply]
was actually using `uv 0.4.9 (77d278f68 2024-09-10`

**Reply by mbway** [reply]
the way versions are ordered, `0.4.9` is older than `0.4.25`. I have a PR that improves the error message https://github.com/PyO3/maturin/pull/2389
